redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -411,6 +416,122 @@ export class RecommendationProvider {
     }
   }
 
+  /**
+  * Check search indexes recommendation
+   * @param redisClient
+   * @param keys
+   * @param client
+   */
+  // eslint-disable-next-line
+  async determineSearchIndexesRecommendation(
+    redisClient: Redis,
+    keys: Key[],
+    client: any,
+  ): Promise<Recommendation> {
+    try {
+      if (client.isCluster) {
+        let processedKeysNumber = 0;
+        let isJSONOrHash = false;
+        let sortedSetNumber = 0;
+        while (
+          processedKeysNumber < keys.length
+          && !isJSONOrHash
+          && sortedSetNumber <= sortedSetCountForCheck
+        ) {
+          if (keys[processedKeysNumber].type !== RedisDataType.ZSet) {
+            processedKeysNumber += 1;
+          } else {
+            let keyType: string;
+            const sortedSetMember = await redisClient.sendCommand(
+              new Command('zrange', [keys[processedKeysNumber].name, 0, 0], { replyEncoding: 'utf8' }),
+            ) as string[];
+            try {
+              keyType = await redisClient.sendCommand(
+                new Command('type', [sortedSetMember[0]], { replyEncoding: 'utf8' }),
+              ) as string;
+            } catch (err) {
+              if (err && checkRedirectionError(err)) {
```

Comment:
Are you sure that we need this?
As I remember we can simplify this "redirection" functionality a lot.
Let's use not node client but `parent client` (general cluster client) to execute commands. In this case all commands will be routed to needed node automatically, no? did you check this behaviour?